### PR TITLE
Enable ObjectFormatterTests

### LINKED ISSUE #15860: Enable more scripting tests on Unix
Search for issue number in CSharpScriptingTests.

### PR DESCRIPTION
**Pull request by sharwell**
Fixes #15860 
See #19027

### CODE REVIEW

**Comment by sharwell** [comment]
@dotnet-bot retest this please

**Comment by sharwell** [comment]
@dotnet-bot retest this please

File: src/Scripting/CSharpTest/ObjectFormatterTests.cs [hunk 1]
```diff
@@ -950,13 +953,11 @@ public void StackTrace_Dynamic()
...
             catch (Exception e)
             {
-                const string filePath = @"z:\Fixture.cs";
-
                 var expected =
 $@"'object' does not contain a definition for 'x'
   + System.Dynamic.UpdateDelegates.UpdateAndExecuteVoid1<T0>(System.Runtime.CompilerServices.CallSite, T0)
-  + Microsoft.CodeAnalysis.CSharp.Scripting.Hosting.UnitTests.ObjectFormatterTests.Fixture2.MethodDynamic(){string.Format(ScriptingResources.AtFileLine, filePath, 10123)}
-  + Microsoft.CodeAnalysis.CSharp.Scripting.Hosting.UnitTests.ObjectFormatterTests.StackTrace_Dynamic(){string.Format(ScriptingResources.AtFileLine, filePath, 10132)}
+  + Microsoft.CodeAnalysis.CSharp.Scripting.Hosting.UnitTests.ObjectFormatterTests.Fixture2.MethodDynamic(){(s_supportsLineNumbers ? string.Format(ScriptingResources.AtFileLine, StackTraceSourceFile, 10124) : "")}
```
Comment (sharwell) [inline src/Scripting/CSharpTest/ObjectFormatterTests.cs]: 💭 The result is different for this test on **ubuntu_16_debug_prtest** (produces line 10123) and **windows_coreclr_test_prtest** (produces line 10124).

File: src/Scripting/CSharpTest/ObjectFormatterTests.cs [hunk 2]
```diff
@@ -13,14 +13,24 @@
 using Microsoft.CodeAnalysis.Scripting.Hosting.UnitTests;
 using ObjectFormatterFixtures;
 using Roslyn.Test.Utilities;
+using Roslyn.Utilities;
 using Xunit;
 
 namespace Microsoft.CodeAnalysis.CSharp.Scripting.Hosting.UnitTests
 {
+    extern alias Scripting;
+    using CoreClrShim = Scripting::Microsoft.CodeAnalysis.CoreClrShim;
```
Comment (jaredpar) [inline src/Scripting/CSharpTest/ObjectFormatterTests.cs]: All using should be outside the namespace. Same with extern. That's the default we use.

File: src/Scripting/CSharpTest/ObjectFormatterTests.cs [hunk 3]
```diff
@@ -13,14 +13,24 @@
...
+
     public class ObjectFormatterTests : ObjectFormatterTestBase
     {
+        /// <seealso href="https://github.com/dotnet/roslyn/issues/19027">CoreCLR scripting tests loading PDBs fail after move to SDK</seealso>
+        private static readonly bool s_supportsLineNumbers = PlatformInformation.IsUnix || CoreClrShim.IsRunningOnCoreClr;
+
         private static readonly ObjectFormatter s_formatter = new TestCSharpObjectFormatter();
 
+        private static string StackTraceSourceFile =>
+            PlatformInformation.IsWindows ? @"z:\Fixture.cs" : "/_/src/Scripting/CSharpTest/z:/Fixture.cs";
```
[the remark this answers is not in the document]
Comment (jaredpar) [reply src/Scripting/CSharpTest/ObjectFormatterTests.cs]: The path of where the tests are running?

File: src/Scripting/CSharpTest/ObjectFormatterTests.cs (hunk 2 above)
Comment (sharwell) [reply src/Scripting/CSharpTest/ObjectFormatterTests.cs]: ➡️ It is not possible to resolve this ambiguity with the standard code style for this project.

File: src/Scripting/CSharpTest/ObjectFormatterTests.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (sharwell) [reply src/Scripting/CSharpTest/ObjectFormatterTests.cs]: I tried using `Path.GetFullPath("z:/Fixture.cs")` on Linux, but it was some other path under `/mnt/...`.

File: src/Scripting/CSharpTest/ObjectFormatterTests.cs (hunk 2 above)
Comment (jaredpar) [reply src/Scripting/CSharpTest/ObjectFormatterTests.cs]: How does moving the extern alias outside the namespace have any impact here?

**Comment by sharwell** [comment]
@agocke I'd like to reopen this and retarget to master, but I need the dev15.7.x branch restored first

**Pull request closed without merging**